File: src/states/localfiles.ts
import { atom } from 'jotai'
import { type Fetchmeta } from './type'

export type LocalfileItem = {
  name: string;
}
export type LocalfilesResponse = {
  items: LocalfileItem[];
}
export type LocalfilesArg = {}

export const localfilesMeta: Fetchmeta<LocalfilesResponse, LocalfilesArg> = {
  name: 'localfiles',
  atom: atom<LocalfilesResponse | null>(null),
}

File: src-tauri/commands/localfiles.rs
use std::env;
use serde::Serialize;

#[derive(Debug, Serialize)]
struct LocalfileItem {
    name: String,
}
#[derive(Debug, Serialize)]
pub struct LocalfilesResponse {
    items: Vec<LocalfileItem>,
}

#[tauri::command]
pub fn localfiles() -> LocalfilesResponse {
    let mut res = LocalfilesResponse { items: Vec::new() };

    let dir = env::current_dir().unwrap().parent().unwrap().to_owned();
    for file in dir.read_dir().unwrap() {
        res.items.push(LocalfileItem {
            name: file.unwrap().path().to_str().unwrap().to_string()
        });
    };
    println!("{:?}", res);
    res
}
